### PR TITLE
[Part 2] Extract TKG client lib to top level package

### LINKED ISSUE #3241: Extract TKG client lib to top level package
**Describe the feature request**
The core functionality of the TKG client lib supports operations used by the cluster and management cluster plugin. To support reuse of logic in this client lib, move it to a separate top level package with its own go module.

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- (x) APIs
- (x) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- (x) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by Vandy-P**
Moving pkg/v1/tkg as a top level go module
Co-authored-by: Sathyanarayanan Saravanamuthu [sathyanarays@.vmware.com](mailto:sathyanarays@.vmware.com), Vandana Pathak [pathakv@vmware.com](mailto:pathakv@vmware.com) , Shashank Girish [girishsh@vmware.com](mailto:girishsh@vmware.com)

### What this PR does / why we need it
The core functionality of the TKG client lib supports operations used by the cluster and management cluster plugin. To support reuse of logic in this client lib, move it to a separate top level package with its own go module.
### Which issue(s) this PR fixes

Fixes #3241 

Change details
Moved all the directories from pkg/v1/tkg to tkg module:

### CODE REVIEW

File: config/crd/bases/config.tanzu.vmware.com_clientconfigs.yaml [hunk 1]
```diff
@@ -32,15 +32,17 @@ spec:
                 description: CLI options specific to the CLI.
                 properties:
                   bomRepo:
-                    description: BOMRepo is the root repository URL used to resolve
-                      the compatibiilty file and bill of materials. An example URL
-                      is projects.registry.vmware.com/tkg.
+                    description: 'Deprecated: BOMRepo has been deprecated and will
+                      be removed from future version BOMRepo is the root repository
+                      URL used to resolve the compatibiilty file and bill of materials.
+                      An example URL is projects.registry.vmware.com/tkg.'
```
Comment (tenczar) [inline config/crd/bases/config.tanzu.vmware.com_clientconfigs.yaml:38]: Is this really supposed to be part of this change?

**Review by tenczar — APPROVED** [review]
LGTM. Out of curiosity, does make generate-fakes still work with this refactor work? I assume the path changes in the fake files came from an idea refactor/move. Once this is finished we need to make sure that make generate-fakes still works.

File: config/crd/bases/config.tanzu.vmware.com_clientconfigs.yaml [hunk 2]
```diff
@@ -32,15 +32,17 @@ spec:
                 description: CLI options specific to the CLI.
                 properties:
                   bomRepo:
-                    description: BOMRepo is the root repository URL used to resolve
-                      the compatibiilty file and bill of materials. An example URL
-                      is projects.registry.vmware.com/tkg.
+                    description: 'Deprecated: BOMRepo has been deprecated and will
+                      be removed from future version BOMRepo is the root repository
```
Comment (maralavi) [inline config/crd/bases/config.tanzu.vmware.com_clientconfigs.yaml:36]: Why there is no new line after deprecation message?

**Review by anujc25 — COMMENTED** [review]
Looks good overall except few existing comments made by @tenczar and @maralavi

**Review by anujc25 — APPROVED** [review]
Changes looks good overall.

Here are the few things:
- We should file an issue to remove `tkrtypes` package dependency that is copied here
- `tkg/buildinfo.IsOfficialBuild` needs to be configured at build time

(We can do this changes as a follow up PR if there are no other blocking comments on this PR)

**Review by tenczar — APPROVED** [review]
LGTM

**Pull request merged**